bin/oneoff/script_clone_command_generator.rb

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env ruby
+
+# Generates the list of commands to clone each unit within an unit group
+# This assumes that the new unit group is already created
+
+require_relative '../../dashboard/config/environment'
+
+def generate_unit_clone_commands(source_unit_group_name, src_year, dstn_year)
+  destination_unit_group_name = source_unit_group_name.sub(src_year, dstn_year)
+
+  puts "\nUnit group: [#{source_unit_group_name}] for destination year [#{dstn_year}]"
+  puts "Create a destination unit group name with name [#{destination_unit_group_name}] at https://levelbuilder-studio.code.org/courses/new and run the commands below"
+
+  UnitGroup.find_by_name(source_unit_group_name).default_units.each do |unit_to_clone|
+    source_unit_name = unit_to_clone.name
+    destination_unit_name = source_unit_name.sub(src_year, dstn_year)
+    puts "Unit.find_by_name('#{source_unit_name}').clone_migrated_unit('#{destination_unit_name}', destination_unit_group_name: '#{destination_unit_group_name}', new_level_suffix: '#{dstn_year}')"
+  end
+end
+
+puts "\nThis script does not generate correct commands for deep learning courses, courses that belong to a course family and those which aren't versioned."
```

Comment:
I'm curious if this is a contained list that can be checked by the script instead of leaving it up to the person executing to read this line.

Reply:
I think that if a course is not versioned, it doesn't have a year stored. You could have the script check each course it pulls and throw an error if there is no year. As for the others, you can definitely have the script check whether there is a course family, and if the deep learning courses are part of a family, that would make it easy to check for those. Otherwise, you could check by a list of hardcoded names.

Reply:
Oh yeah, that's a good suggestion. Will do that, and keep the warning though just to be safe.

Reply:
yes, it should be possible for your program to determine these attributes by examining the script/unit. I do not feel super comfortable with the idea of the caller of this program needing to read a set of instructions to know whether this program can be safely run -- it seems like a good minimum bar to get this into the repo would be for it to either handle the input correctly and output the correct syntax, or else give an error saying that it cannot handle that situation.

Reply:
I spot checked a few courses and using the family_name attribute to check for nil seemed to cover all these cases (unversioned stand alone unit and deep learning courses). Updated the check to skip generating clone commands for those cases.

Reply:
Adding sample output below

```
This script does not generate correct commands for deep learning courses, courses that belong to a course family and those which aren't versioned.
Pull latest from staging and run seeding to ensure the local curriculum state matches that of levelbuilder environment.

Enter the source year:2023
Enter the destination year:2024
Enter a comma seperated list of scripts to clone:dlp21-csd-overview,self-paced-pl-csd-unit4-2023
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
Generating commands for unit: [dlp21-csd-overview] for destination year [2024]
Clone commands for units that are not part of a family aren't supported yet. Please check the how-to guide and consider adding the support.

Unit group: [self-paced-pl-csd-unit4-2023] for destination year [2024]
Create a destination unit group name with name [self-paced-pl-csd-unit4-2024] at https://levelbuilder-studio.code.org/courses/new and run the commands below
Unit.find_by_name('self-paced-pl-csd-unit4-1-2023').clone_migrated_unit('self-paced-pl-csd-unit4-1-2024', destination_unit_group_name: 'self-paced-pl-csd-unit4-2024', new_level_suffix: '2024')
Unit.find_by_name('self-paced-pl-csd-unit4-2-2023').clone_migrated_unit('self-paced-pl-csd-unit4-2-2024', destination_unit_group_name: 'self-paced-pl-csd-unit4-2024', new_level_suffix: '2024')
```